integrations/cohere/src/haystack_integrations/components/embedders/cohere/document_embedder.py

```diff
@@ -175,98 +164,65 @@ def run(self, documents: List[Document]):
-        if api_key is None:
-            err_msg = (
-                "Cohere API key is missing. Pass it via the `api_key` or set the "
-                "`COHERE_API_KEY` or `CO_API_KEY` environment variable."
-            )
-            raise ValueError(err_msg)
-
-        if self.use_async_client:
-            cohere_client = AsyncClientV2(
-                api_key,
-                base_url=self.api_base_url,
-                timeout=self.timeout,
-                client_name="haystack",
-            )
-            all_embeddings, metadata = asyncio.run(
-                get_async_response(
-                    cohere_client, texts_to_embed, self.model, self.input_type, self.truncate, self.embedding_type
-                )
-            )
-        else:
-            cohere_client = ClientV2(
-                api_key,
-                base_url=self.api_base_url,
-                timeout=self.timeout,
-                client_name="haystack",
-            )
-            all_embeddings, metadata = get_response(
-                cohere_client,
-                texts_to_embed,
-                self.model,
-                self.input_type,
-                self.truncate,
-                self.batch_size,
-                self.progress_bar,
-                self.embedding_type,
-            )
+        all_embeddings, metadata = get_response(
+            self._client,
+            texts_to_embed,
+            self.model,
+            self.input_type,
+            self.truncate,
+            self.batch_size,
+            self.progress_bar,
+            self.embedding_type,
+        )
 
         for doc, embeddings in zip(documents, all_embeddings):
             doc.embedding = embeddings
 
         return {"documents": documents, "meta": metadata}
 
     @component.output_types(documents=List[Document], meta=Dict[str, Any])
-    async def run_async(self, documents: List[Document]):
+    async def run_async(self, documents: List[Document]) -> Dict[str, Union[List[Document], Dict[str, Any]]]:
         """
-        Asynchronously embed list of documents.
-
-        :param documents:
-            List of documents to embed.
+        Embed a list of `Documents` asynchronously.
 
-        :returns:
-            A dictionary with the following keys:
-            - `documents`: A list of documents with embeddings.
-            - `meta`: Information about the usage of the model.
-
-        :raises TypeError:
-            If the input is not a list of documents.
+        :param documents: documents to embed.
+        :returns:  A dictionary with the following keys:
+            - `documents`: documents with the `embedding` field set.
+            - `meta`: metadata about the embedding process.
+        :raises TypeError: if the input is not a list of `Documents`.
         """
 
-        documents = self._prepare_input(documents)
+        if not isinstance(documents, list) or (documents and not isinstance(documents[0], Document)):
```

Comment:
Just a comment: If there is more validation, we could extract this into a function that we can reuse in `run` and `run_async`. Right now code redundancy is minimal so I am okay with keeping it as is.